packages/graph/hash_graph/lib/graph/src/store/postgres/query/mod.rs

```diff
@@ -13,8 +13,11 @@ use std::fmt::{self, Formatter};
 pub use self::{
     condition::Condition,
```

Comment:
(comment not for this line)
I have a pretty crucial thing we have to take care of for our queries regarding `QueryPath::Custom`. This would be taking in user input (as far as I understand) and putting it into parts of the query for selection.
A malicious filter like:

```rust
let filter_a: Filter<DataType> = Filter::Any(vec![Filter::Equal(
  Some(FilterExpression::Path(DataTypeQueryPath::Custom(
      Cow::Borrowed("' = ''; -- \n select * from entities; \n --"),
  ))),
  Some(FilterExpression::Path(DataTypeQueryPath::Custom(
      Cow::Borrowed("right"),
  ))),
)]);
```

Could generate a query that looks like 

```sql
select ...
WHERE ("data_types"."schema"->>'' = ''); -- 
 select * from entities; 
 --' = "data_types"."schema"->>'right')
```

We should definitely figure out what we do about this.

Reply:
cc: @Alfred-Mountfield @benwerner01

Reply:
I just noticed the same issue for the `properties` field in #1228. I see two possible solutions and I prefer the first:
- make `key` in `Cow(key)` and `property` in `Properties(property)` a prepared parameter, however, I need to test if this allowed for Json here
- Make sure, every `'` is escaped by `\` and `\` by `\\` so it's not possible to escape the path. For now, this is not an issue as the queries are not exposed, but we should take care of this before exposing it to the REST endpoints.

Reply:
I would be worried about  trying to sanitize the path ourselves, it's difficult to do it properly, and I think there are many examples of it going wrong. 
That being said, we are limiting the custom fields to values inside accessors as you're pointing out. We might need to run some SQL injection fuzzer if we do implement something ourselves :sweat_smile:

Reply:
`"entities"."properties"->>$1` is working fine. I will add a fix in a [follow-up task](https://app.asana.com/0/0/1203205825668112/f) (_internal_)

Reply:
That's really great, makes it so much easier to handle :rocket:

Reply:
Fixed in https://github.com/hashintel/hash/pull/1230